### ISSUE TITLE
Automerge: Keep message size below 4096 bytes to not exceed gossip broadcast limit 

### ISSUE THREAD

**Issue by adzialocha** [p2panda/aardvark#11]
Automerge gives us fairly large payloads which break the default gossip message size limit by `iroh-gossip` (4092 bytes) even when using `safe_incremental`.

This especially happens if another peer edits a document for the first time which already contains some text, even if it's just adding one single character. It's also surprising that the 4kb limit is reached even if the text itself is less than ca. 100 characters long.

I believe we can fix this by understanding better how Automerge's "diffs" are made and possibily using more low-level methods of their library to really only send the actual changed text.

**Comment by adzialocha** [comment]
I've bumped the limit to some arbitrary number for now to work around this issue until we looked into Automerge: https://github.com/p2panda/aardvark/commit/69a0d988405d4d6f9adc153c5fed4a3d0a58118f#diff-87b01ee53083ffe834159a7e4e9d68383510284bf7c17dacd1abe13e52306a4dR130

**Comment by swick** [comment]
Can't you chop up the message internally somehow? Requiring us to send messages with a certain size to the "networking" part of the app seems a bit strange.

**Comment by adzialocha** [comment]
> Can't you chop up the message internally somehow? Requiring us to send messages with a certain size to the "networking" part of the app seems a bit strange.

Yes, that makes sense if the payload is simply just large (like, very large text documents), but 4kb data _is_ already a lot of text and that it's caused by 100 characters smells more like a bug / wrong use of the Automerge library. So before we dig into optimizations like breaking the network messages apart I'd make sure we're using Automerge "correctly" first.

**Comment by adzialocha** [comment]
My theory is that automerge can't distinct between authors when calculating the diffs. It takes all the edits of the other peer and adds them into the one of the newly arriving peer, merging the whole edit history of peer A to the small edit (adding one character) of peer B.

**Comment by adzialocha** [comment]
EtherSync applies their own diffing algorithm to identify which parts of the text have _really_ been changed, I assume that's required as Automerge can't figure it out itself? https://github.com/ethersync/ethersync/blob/main/daemon%2Fsrc%2Fdocument.rs#L136-L155

**Comment by adzialocha** [comment]
This can be closed now as we've replaced Automerge with Loro CRDT (where this problem did not show up).

We still need to take care of this of course: https://github.com/p2panda/aardvark/issues/24
